webapp/components/FilterPosts/FilterPostsMenuItems.vue

```diff
@@ -0,0 +1,221 @@
+                :icon="category.icon"
+                :primary="isActive(category.id)"
+                @click.stop.prevent="toggleCategory(category.id)"
+              />
+              <ds-space margin-bottom="small" />
+            </ds-flex-item>
+            <ds-flex>
+              <ds-flex-item>
+                <label>{{ category.name }}</label>
+              </ds-flex-item>
+              <ds-space margin-bottom="xx-large" />
+            </ds-flex>
+          </ds-flex>
+        </ds-flex>
+      </ds-flex-item>
+      <ds-flex-item :width="{ base: '50%', sm: '0%', md: '50%', lg: '10%' }">
+        <ds-flex v-for="category in chunk[6]" :key="category.id" class="categories-list">
+          <ds-flex>
+            <ds-flex-item width="100%">
+              <ds-button
+                :icon="category.icon"
+                :primary="isActive(category.id)"
+                @click.stop.prevent="toggleCategory(category.id)"
+              />
+              <ds-space margin-bottom="small" />
+            </ds-flex-item>
+            <ds-flex>
+              <ds-flex-item>
+                <label>{{ category.name }}</label>
+              </ds-flex-item>
+              <ds-space margin-bottom="xx-large" />
+            </ds-flex>
+          </ds-flex>
+        </ds-flex>
+      </ds-flex-item>
+      <ds-flex-item :width="{ base: '50%', sm: '0%', md: '50%', lg: '10%' }">
+        <ds-flex v-for="category in chunk[7]" :key="category.id" class="categories-list">
+          <ds-flex>
+            <ds-flex-item width="100%">
+              <ds-button
+                :icon="category.icon"
+                :primary="isActive(category.id)"
+                @click.stop.prevent="toggleCategory(category.id)"
+              />
+              <ds-space margin-bottom="small" />
+            </ds-flex-item>
+            <ds-flex>
+              <ds-flex-item>
+                <label>{{ category.name }}</label>
+              </ds-flex-item>
+              <ds-space margin-bottom="xx-large" />
+            </ds-flex>
+          </ds-flex>
+        </ds-flex>
+      </ds-flex-item>
+    </ds-flex>
+  </ds-container>
+</template>
+<script>
+export default {
+  props: {
+    chunk: { type: Array, default: () => [] },
+  },
+  data() {
+    return {
+      selectedCategoryIds: [],
+      allCategories: true,
+    }
+  },
+  methods: {
+    isActive(id) {
+      const index = this.selectedCategoryIds.indexOf(id)
+      if (index > -1) {
+        return true
+      }
+      return false
+    },
+    toggleCategory(id) {
+      const index = this.selectedCategoryIds.indexOf(id)
+      if (index > -1) {
+        this.selectedCategoryIds.splice(index, 1)
+      } else {
+        this.selectedCategoryIds.push(id)
+      }
+      if (!id) this.selectedCategoryIds = []
+      this.selectedCategoryIds.length ? (this.allCategories = false) : (this.allCategories = true)
+      this.$emit('filterPosts', this.selectedCategoryIds)
+    },
+  },
+}
+</script>
```

Comment:
maybe this is an improvement @roschaefer @Tirokk @ulfgebhardt @ogerly 
I think it might actually add lines of code, but it extracts the menu items into their own component